agent_issuance/src/credential/command.rs

```diff
@@ -7,13 +7,16 @@ use super::entity::Data;
 #[serde(untagged)]
 pub enum CredentialCommand {
     CreateUnsignedCredential {
+        credential_id: String,
```

Comment:
Is the caller now _required_ to pass in an identifier for the credential?

Reply:
Generate it?

Reply:
This is not possible with the `cqrs` framework we're using because the underlying `CqrsFramework::execute` method requires the aggregate ID to already exist before executing the Command: https://github.com/serverlesstechnology/cqrs/blob/main/src/cqrs.rs#L124-L131

Reply:
Ah, my main concern was that the caller has to generate and pass in the ID themselves, but I just found that it's still generated inside the application (`agent_api_rest`). 👍